src/Redux/bookmarkSlice.js

```diff
@@ -1,26 +1,43 @@
-import {createSlice} from '@reduxjs/toolkit';
+import { createSlice } from "@reduxjs/toolkit";
 
-//redux store for boookmarks
+//redux store for bookmarks
 
 const initialState = {
-  sections: [],
+  bookmarkArray: [
+    //TEST data
+    // {
+    //   legislation: "CrimCode",
+    //   sectionNum: "424.1(b)",
+    //   partLabel: "PART X",
+    //   sectionHeading:
+    //     "Breach of Contract, Intimidation and Discrimination Against Trade Unionists",
+    // },
+  ],
 };
 
 const bookmarkSlice = createSlice({
-  name: 'bookmarks',
+  name: "bookmarks",
   initialState,
   reducers: {
     addBookmark: (state, action) => {
-      state.sections.push(action.payload);
+      state.bookmarkArray.push(action.payload);
     },
     removeBookmark: (state, action) => {
-      state.sections = state.sections.filter(
-        sections => sections.section !== action.payload.section,
+      state.bookmarkArray = state.bookmarkArray.filter(
+        bookmarkItem =>
+          !(
+            bookmarkItem.legislation == action.payload.legislation &&
```

Comment:
Can you document these?

Reply:
Done

Reply:
Why is it that adding needs 5 items to match but only 4 to remove?

Reply:
Future proofing.